src/Service/Startup.cs

```diff
@@ -153,8 +160,11 @@ public void ConfigureServices(IServiceCollection services)
 
             //Enable accessing HttpContext in RestService to get ClaimsPrincipal.
             services.AddHttpContextAccessor();
-
-            ConfigureAuthentication(services, configProvider);
+            services.AddTransient(implementationFactory: (serviceProvider) =>
```

Comment:
interesting, what does `AddTransient` do?

Reply:
I think this section needs to be approached differently. I think the intent is to react to a new config file after startup and configure services (and consequently the DI container) is already created. 

This PR aims to address hot reloading the runtime section of the runtime config. So that means any new config which changes up the authentication mechanism, will need to have a different authentication scheme configured. 

Due to how authentication is configured only at startup, we only have one chance to configure it. I think we'll need to refactor the authN setup section to register all authentication schema unconditionally and allow the mechanism which "authenticate" a request, to dynamically determine which authN scheme to choose based on runtime configuration settings. 

```csharp
services.AddAuthentication("<no default or default SWA>")
    .AddEasyAuthAuthentication(easyAuthAuthenticationProvider: EasyAuthType.AppService)
    .AddEasyAuthAuthentication(easyAuthAuthenticationProvider: EasyAuthType.StaticWebApps)
    .AddSimulatorAuthentication()
    .AddJwtBearer("<use ioptionsmonitor here because jwt settings may change in new config files>");
```

To then register changes to jwtbeareroptions, this asp.net repo thread provides suggestions on how to monitor the jwt options changes and signal that to the underlying authentication config: https://github.com/dotnet/aspnetcore/issues/49586#issuecomment-1671838595

Reply:
Then in `ClientRoleHeaderAuthenticationMiddleware.InvokeAsync(...)` all calls to `httpContext.AuthenticateAsync(..)` and `httpContext.ChallengeAsync(...)` need to use the method signature that requires you to pass in the authetnication scheme. The authentication scheme can be dynamically resolved by helper methods on the runtime config/provider class.

Reply:
i see what you mean with how we do the authentication configuring only at startup. Let's sync up on this one, and thanks for pointing this out!

Reply:
Just to update the decisions here. After syncing with Sean, we've determined a plan for updating the authentication options, and this new design will be added to the related design document, and then come in the future iterations on hot reload.